concert/app/src/functions/on_event_stage_changed/index.py

```diff
@@ -35,21 +34,65 @@ def handler(event, context):
     return {"statusCode": 204}
 
 
-def update_stage_record(message):
+def update_event_stage_and_song_list(message):
     data = copy(message["data"])
-    stage_id = data.pop(ATTR_EVENT_STAGE_ID)
-
-    update_args = {
-        "Key": {ATTR_RECORD_ID: RECORD_ID_EVENT_STAGE},
-        "UpdateExpression": "SET #stage_id = :stage_id",
-        "ExpressionAttributeNames": {"#stage_id": ATTR_EVENT_STAGE_ID},
-        "ExpressionAttributeValues": {":stage_id": stage_id},
-        "ReturnValues": "NONE",
+
+    event_stage_put_transaction_item = get_event_stage_put_transaction_item(data)
+    song_list_update_transaction_item = get_song_list_update_transaction_item(data)
+
+    transaction_items = [event_stage_put_transaction_item]
+    if song_list_update_transaction_item:
+        transaction_items.append(song_list_update_transaction_item)
+
+    return dynamodb.transact_write_items(TransactItems=transaction_items)
+
+
+def get_event_stage_put_transaction_item(message_data):
+    stage_id = message_data.pop(ATTR_EVENT_STAGE_ID)
+
+    event_stage_update_args = {
+        "Item": {
+            ATTR_RECORD_ID: {"S": RECORD_ID_EVENT_STAGE},
+            ATTR_EVENT_STAGE_ID: {"S": stage_id},
+        },
+        "TableName": DYNAMODB_TABLE_NAME,
     }
 
-    for key, value in data.items():
-        update_args["UpdateExpression"] += ", #{} = :{}".format(key, key)
-        update_args["ExpressionAttributeNames"]["#{}".format(key)] = key
-        update_args["ExpressionAttributeValues"][":{}".format(key)] = value
+    for key, value in message_data.items():
+        if isinstance(value, int):
```

Comment:
I don't believe that we have any non-int or string values planned for the event stage record, but if we do, we may need to update this type check.

Reply:
You can do this using boto3:
```py
serializer = boto3.dynamodb.types.TypeSerializer()
event_stage_update_args["Item"] = {k: serializer.serialize(v) for k,v in message_data.items()}
```

Reply:
Good suggestion. I feel much better about that. Done.